core/imt/inc/ROOT/RSlotStack.hxx

```diff
@@ -18,10 +18,15 @@
 namespace ROOT {
 namespace Internal {
 
-/// This is an helper class to allow to pick a slot resorting to a map
-/// indexed by thread ids.
-/// WARNING: this class does not work as a regular stack. The size is
-/// fixed at construction time and no blocking is foreseen.
+/// A thread-safe stack of N indexes (0 to size - 1).
```

Comment:
```suggestion
/// A thread-safe, constant-size stack of N indexes (0 to size - 1).
```

Reply:
the stack is not constant size. we pop slot indexes from it and then put them back. after construction, there are `size` indexes available and it's UB to call `GetSlot` more than `size` times without calling `ReturnSlot` in the meanwhile between.